### ISSUE TITLE
Regent: Inconsistencies in behavior of futures

### ISSUE THREAD

**Issue by manopapad** [StanfordLegion/legion#471]
Run this code through Regent:

```
import "regent"

task foo()
  return 0.0
end

task bar(x : double)
  regentlib.c.printf('%e\n', x)
end

task main()
  -- this forces a __future_get_result
  var a = 1.0/foo()
  bar(a)
  -- this doesn't
  var b = foo()/1.0
  bar(b)
end
main:printpretty()
```

You will get this output:

```
task main()
-- leaf (false), inner (true), idempotent (true), replicable (true)
  var $22 : future(double) = foo()
  var $a : double = __future_get_result((1/$22))
  bar($a)
  var $21 : future(double) = foo()
  var $b : future(double) = ($21/1)
  bar($b)
end
```

It seems like the compiler treats the two cases of division differently.

**Comment by manopapad** [comment]
Output of -fpretty 1:

```
task foo() : double
-- leaf (true), inner (false), idempotent (true), replicable (true)
  return 0
end

task bar($x : double)
-- leaf (true), inner (false), idempotent (false), replicable (false)
  printf("%e\n", $x)
end

task main()
-- leaf (false), inner (true), idempotent (true), replicable (true)
  var $20 : future(double) = foo()
  var $a : double = __future_get_result((1/$20))
  bar($a)
  var $21 : future(double) = foo()
  var $b : future(double) = ($21/1)
  bar($b)
end
```

**Comment by elliottslaughter** [comment]
Should be fixed in 8526a218d3b64b9b33bd49eb3b1c72e45a5fa7dc.

TL;DR: It's not a good idea to use `nil` as an argument to a variadic function in Lua. If you want to iterate the arguments with `ipairs`, you'll miss some values if `nil` comes in the middle.

**Comment by elliottslaughter** [comment]
Sorry, silly mistake. Please try d3de7ee0686a04f11b0095ef454b79ef2d09dbfb.

**Comment by manopapad** [comment]
That worked!
